lib/experimental/class-wp-html-tokenizer-2.php

```diff
@@ -0,0 +1,492 @@
+<?php
+
+class WP_Attribute_Match {
+	private $name;
+	private $value;
+	private $start_index;
+	private $end_index;
+
+	/**
+	 * @param $name
+	 * @param $value
+	 * @param $start_index
+	 * @param $end_index
+	 */
+	public function __construct( $name, $value, $start_index, $end_index ) {
+		$this->name        = $name;
+		$this->value       = $value;
+		$this->start_index = $start_index;
+		$this->end_index   = $end_index;
+	}
+
+	public function __toString() {
+		return "{$this->name}=\"{$this->value}\"";
```

Comment:
clever creating the attribute match class. although this is a detail and it's probably not the right time, remember we need to escape the double-quotes and `>` inside this

Reply:
Thanks! This toString is mostly for debugging and I don't intend to include it in the final version at the moment.